### PR TITLE
Attempt to fix feedstock

### PR DESCRIPTION
**Pull request by Olllom**
See discussion https://github.com/conda-forge/staged-recipes/pull/11679

Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Olllom** [comment]
@conda-forge-admin please rerender

**Comment by Olllom** [comment]
@xhochy ready for review

This is a (partial) fix for the libcmaes feedstock, following our discussion in https://github.com/conda-forge/staged-recipes/pull/11679.

Unlike the master of the feedstock, this version builds on osx and linux for the most recent libcmaes version. There is a problem with the python bindings (only on osx), which is also the case for all previous builds that I tested locally. So nothing is worse here compared to the previous version. Since the feedstock is not going to be better than the actual code, I am not going to dive much deeper. This PR certainly improves the feedstock and provides the C++ libraries (osx and linux) and working python bindings (osx only).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,42 +7,39 @@ package:
...
+    - numpy
     - boost
 
 test:
   commands:
     - test -f ${PREFIX}/include/libcmaes/cmaes.h  # [unix]
     - test -f ${PREFIX}/lib/libcmaes${SHLIB_EXT}  # [unix]
   imports:
-    - lcmaes
+    - lcmaes  # [linux]
```
Comment (xhochy) [inline recipe/meta.yaml:42]: Don't skip imports

**Comment by Olllom** [comment]
@jschueller Well...

the previous version does not build on windows, either.
https://github.com/conda-forge/libcmaes-feedstock/blob/c6fbb03f8952355456ff39646acc0de788221e5f/recipe/meta.yaml#L18

and as far as osx goes, I get this locally
```
conda create -n lcmaestest libcmaes python

Collecting package metadata (current_repodata.json): done
Solving environment: done

The following NEW packages will be INSTALLED:

...
  libcmaes           conda-forge/osx-64::libcmaes-0.9.5-py36hf738d87_3
...


conda activate lcmaestest

python -c "import lcmaes"  
[1]    98368 segmentation fault  python -c "import lcmaes"
```

So, what is actually worse than on the master?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,42 +7,39 @@ package:
...
-    # https://github.com/beniz/libcmaes/pull/149
-    - py3k.patch
+  sha256: 1b23e53eb4acb123348e7d00dfdd7c7699c513950347fcde7425376ad333f6c5
 
 build:
-  number: 3
+  number: 0
   # does not build with vc9
-  skip: true  # [win and py<35]
+  skip: true  # [win or py<35]
```
Comment (xhochy) [inline recipe/meta.yaml:15]: What is the issue with Windows?

**Comment by xhochy** [comment]
> the previous version does not build on windows, either.
> https://github.com/conda-forge/libcmaes-feedstock/blob/c6fbb03f8952355456ff39646acc0de788221e5f/recipe/meta.yaml#L18

This only skips for VS2008/vc8, not VS2015/vc14 though.

**Comment by jschueller** [comment]
it segfaults on osx because it links with python libs.
Python libs should only be linked on windows. On osx we have to link with "-undefined dynamic_lookup"
A patch should be submitted in that sense.

**Comment by Olllom** [comment]
@xhochy my bad, thanks. 
So I guess this is not ready to be merged then.

@jschueller Thanks. Do you also have an idea of what's wrong with the windows build? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=166446&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=24939

**Comment by jschueller** [comment]
for windows i dont know what happens

File: recipe/meta.yaml (hunk 2 above)
Comment (Olllom) [reply recipe/meta.yaml]: I have no idea. The osx build is working fine now after patching in the dynamic lookup as proposed by @jschueller. Does the MSVC linker also need a similar dynamic lookup flag? I have no idea why the Windows tests are failing.

File: recipe/meta.yaml (hunk 1 above)
Comment (Olllom) [reply recipe/meta.yaml]: OK

File: recipe/meta.yaml (hunk 2 above)
Comment (jschueller) [reply recipe/meta.yaml]: no, see some tests are succeeding, so the module should import ok
so it's something else

File: recipe/meta.yaml [hunk 3]
```diff
@@ -7,34 +7,33 @@ package:
 source:
   fn: libcmaes-{{ version }}.tar.gz
   url: https://github.com/beniz/libcmaes/archive/{{ version }}.tar.gz
-  sha256: 899fcc44b551e4976a77e48e41bdd5dafc3394bd3fcb3c55996c02e4267bfa28
+  sha256: 1b23e53eb4acb123348e7d00dfdd7c7699c513950347fcde7425376ad333f6c5
   patches:
-    # https://github.com/beniz/libcmaes/pull/149
-    - py3k.patch
+    - dynamic_lookup.patch  # [osx]
```
Comment (jschueller) [inline recipe/meta.yaml:13]: the logic should be: if windows -> link to python libs else if osx -> undefined lookup (on linux no need to do anything)
their lnk_python flag can be removed
this patch should be upstreamed then

File: recipe/dynamic_lookup.patch [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ python_add_module (lcmaes lcmaes.cc)
++
+ target_link_libraries(lcmaes cmaes ${Boost_LIBRARIES})
+-if (LINK_PYTHON)
++
++if (WIN32)
+   target_link_libraries(lcmaes ${PYTHON_LIBRARIES})
++elseif (APPLE)
++  set_target_properties(lcmaes PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+ endif ()
```
Comment (Olllom) [inline recipe/dynamic_lookup.patch:29]: @jschueller I guess that is what you meant?

Comment (jschueller) [reply recipe/dynamic_lookup.patch:29]: yes, totally, can you submit it upstream ?

Comment (Olllom) [reply recipe/dynamic_lookup.patch:29]: Will do once this is all merged and final.

File: recipe/meta.yaml (hunk 2 above)
Comment (Olllom) [reply recipe/meta.yaml]: OK, fixed. All platforms are passing.
@xhochy can you give it another look?

File: recipe/build.sh [hunk 5]
```diff
@@ -1,16 +1,13 @@
...
 
-if test "${PY3K}" = "1"
-then
-  with_boost_python3="--with-boost-python=boost_python3"
-fi
-
-./autogen.sh
-./configure --prefix=${PREFIX} --enable-python --enable-onlylib --with-eigen3-include=${PREFIX}/include/eigen3 ${with_boost_python3}
-make -j${CPU_COUNT}
+make
```
Comment (jschueller) [inline recipe/build.sh:11]: you can keep -j${CPU_COUNT}

File: recipe/meta.yaml [hunk 6]
```diff
@@ -7,34 +7,33 @@ package:
...
+  number: 0
   # does not build with vc9
   skip: true  # [win and py<35]
 
 requirements:
   build:
-    - autoconf  # [unix]
-    - automake  # [unix]
+      #- autoconf  # [unix]
+      #- automake  # [unix]
```
Comment (jschueller) [inline recipe/meta.yaml:22]: you can cleanup these

**Pull request merged**